### PR TITLE
Docs: add tutorial for setup

### PR DESCRIPTION
**Pull request by klees**
Hi @acgruber,

I would appreciate some feedback if this is what we are aiming at from the perspective of style and information density. I will try to find someone that looks into this from the perspective of the content.

Kind regards!

### CODE REVIEW

File: docs/development/devguide/tutorial/02-tools/03-setup.md [hunk 1]
```diff
@@ -1,3 +1,119 @@
...
+ILIAS requires to function. Move to the ILIAS folder
+
+```sh
+cd ILIAS
+```
+
+and start by installing the JavaScript dependencies:
+
+```sh
+npm clean-install
```
Comment (PurHur) [inline docs/development/devguide/tutorial/02-tools/03-setup.md]: Please add --ignore-scripts

**Comment by acgruber** [comment]
Hi @klees,

This is exactly what we're aiming for both in terms of style and density. Great example of how to use enganging language to guide a newbie through an article as well. Thanks a lot for this contribution!

Cheers!

File: docs/development/devguide/tutorial/02-tools/03-setup.md (hunk 1 above)
Comment (klees) [reply docs/development/devguide/tutorial/02-tools/03-setup.md]: Done. Thanks!

File: docs/development/devguide/tutorial/02-tools/03-setup.md [hunk 2]
```diff
@@ -1,3 +1,119 @@
...
+
+## What do I Need to Watch Out For? (Dos & Dont's)
+
+* Please be aware that exposing a web application to the internet is not something
+  to be done lightly. This tutorial covers the basics to create a running installation
+  on a local system for development purpose. Make sure to configure your system
+  appropriatly to make sure that no one illegitematly uses your local installation.
+* Certain changes to the code will require to rebuild artifacts. If you e.g. create
+  new GUI classes or install a plugin, parts of the artifacts will need to be
+  refreshed. If in doubt, run `composer install`. This will rebuild static artifacts
```
Comment (tfamula) [inline docs/development/devguide/tutorial/02-tools/03-setup.md:118]: Is there a reason why you are using `composer install` here and not `composer dump-autoload`?

Comment (klees) [reply docs/development/devguide/tutorial/02-tools/03-setup.md:118]: I think `composer install` is more general in the sense that it does all the required activities to get required code in place. People would need to understand what has changed in some code update to understand if `dump-autoload` is enough or `install` is required. Would not want to go there in this tutorial. Is there something where `install` would have a disadvantage over `dump-autoload`?

Comment (tfamula) [reply docs/development/devguide/tutorial/02-tools/03-setup.md:118]: Not really, I was asking more out of curiosity. There are only two really small ones which come to my mind. `composer dump-autoload` is a bit faster and it is not that strict regarding PHP versions. When switching between different ILIAS versions locally, I have to pay more attention using the right PHP version when executing `composer install`. But your argument prevails here of course – thanks for the explanation!

Comment (klees) [reply docs/development/devguide/tutorial/02-tools/03-setup.md:118]: Thanks for the feedback =)

**Comment by acgruber** [comment]
Merged into trunk. Thanks again @klees for the contribution and also @PurHur and @tfamula for your inputs!

**Pull request merged**